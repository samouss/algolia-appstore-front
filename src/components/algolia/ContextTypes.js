import PropTypes from 'prop-types';

export default {
  algoliaClient: PropTypes.object.isRequired,
  algoliaHelper: PropTypes.object.isRequired,
};
