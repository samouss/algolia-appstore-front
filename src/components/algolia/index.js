export { default as Provider } from './Provider';
export { default as withPaginateResults } from './withPaginateResults';
