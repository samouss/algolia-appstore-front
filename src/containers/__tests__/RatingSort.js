import React from 'react';
import { shallow } from 'enzyme';
import { createMockAlgoliaClient, createMockAlgoliaHelper } from 'test/algolia';
import WithQuery, { RatingSort } from '../RatingSort';

describe('<RatingSort />', () => {
  const defaultProps = {
    indexName: 'index',
    onChange: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <RatingSort
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange', () => {
    const props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const component = shallow(
      <RatingSort
        {...props}
      />,
    );

    component
      .find('RadioInput')
      .first()
      .simulate('change', {
        currentTarget: {
          value: 'apps_rating_desc',
        },
      });

    expect(props.onChange).toHaveBeenCalledWith('apps_rating_desc');
  });

  describe('withQuery', () => {
    it('expect to render', () => {
      const context = {
        algoliaClient: createMockAlgoliaClient(),
        algoliaHelper: createMockAlgoliaHelper(),
      };

      const component = shallow(
        <WithQuery />,
        { context },
      );

      expect(component).toMatchSnapshot();
    });
  });
});
