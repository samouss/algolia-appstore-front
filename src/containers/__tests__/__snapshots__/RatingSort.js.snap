// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingSort /> expect to render 1`] = `
<SortBy>
  <RadioInput
    checked={false}
    customClassName=""
    id="sort-desc"
    name="sort"
    onChange={[Function]}
    value="apps_rating_desc"
  >
    Highest rating
  </RadioInput>
  <RadioInput
    checked={false}
    customClassName=""
    id="sort-asc"
    name="sort"
    onChange={[Function]}
    value="apps_rating_asc"
  >
    Lowest rating
  </RadioInput>
</SortBy>
`;

exports[`<RatingSort /> withIndex expect to render 1`] = `
<RatingSort
  indexName="sample"
  onChange={[Function]}
/>
`;
